agent/proxycfg/upstreams.go

```diff
@@ -148,7 +149,17 @@ func (s *handlerUpstreams) handleUpdateUpstreams(ctx context.Context, u UpdateEv
 		passthroughs := make(map[string]struct{})
 
 		for _, node := range resp.Nodes {
-			if !node.Service.Proxy.TransparentProxy.DialedDirectly {
+			dialedDirectly := node.Service.Proxy.TransparentProxy.DialedDirectly
```

Comment:
Can you update this proxycfg test case to validate the change?
https://github.com/hashicorp/consul/blob/main/agent/proxycfg/state_test.go#LL2219

Could update the call to `TestCompileConfigEntries` to include a service-defaults with `DialedDirectly` set to true, and flip one of the CheckServiceNodes to set it to false. That way we check that we still get both addresses as passthrough upstreams.